File: Leet Code/Remove Duplicates from Sorted Array/Remove_duplicates.py
class Solution:
    # Define a method to remove duplicate elements from a list in-place
    def removeDuplicates(self, nums: list[int]) -> int:
        # Convert the list to a set to automatically remove duplicates
        list_set = set(nums)
        
        # Clear the original list
        nums.clear()
        
        # Extend the list with the unique elements from the set
        nums.extend(list_set)
        
        # Sort the list (optional, depends on whether you want the result to be sorted)
        nums.sort()
        
        # Return the length of the modified list (number of unique elements)
        return len(nums)

File: Leet Code/Remove Duplicates from Sorted Array/Remove_duplicates.js
class Solution {
  // Define a method to remove duplicate elements from an array in-place
  removeDuplicates(nums) {
      // Convert the array to a Set to automatically remove duplicates
      const set = new Set(nums);
      
      // Clear the original array
      nums.length = 0;
      
      // Spread the unique elements from the Set into the original array
      nums.push(...set);
      
      // Sort the array (optional, depends on whether you want the result to be sorted)
      nums.sort((a, b) => a - b);
      
      // Return the length of the modified array (number of unique elements)
      return nums.length;
  }
}

// Example usage:
const solution = new Solution();
const inputArray = [1, 2, 3, 2, 4, 5, 6, 5, 7];
const resultLength = solution.removeDuplicates(inputArray);

console.log(inputArray); // Modified array with unique elements
console.log(resultLength); // Number of unique elements in the array
